Jenkinsfile

```diff
@@ -10,19 +10,56 @@ node {
 
   stage 'Build'
 
-  /* Call the maven build. */
-  mvn "clean install -B -V -U -e -Dsurefire.useFile=false -Dmaven.test.failure.ignore=true"
+  /* Call the maven build.  No tests./ */
+  mvn "clean install -B -V -U -e -DskipTests"
+
+  stage 'Test'
+
+  /* Run tests in parallel on multiple nodes */
+  runParallelTests()
 
   /* Save Results. */
   stage 'Results'
 
-  /* Archive the test results */
-  step([$class: 'JUnitResultArchiver', testResults: '**/target/surefire-reports/TEST-*.xml'])
-
   /* Archive the build artifacts */
   step([$class: 'ArtifactArchiver', artifacts: 'target/*.hpi,target/*.jpi'])
 }
 
+void runParallelTests() {
+  /* Request the test groupings.  Based on previous test exection. */
+  def splits = splitTests parallelism: [$class: 'CountDrivenParallelism', size: 4], generateInclusions: true
+
+  def testGroups = [:]
+  for (int i = 0; i < splits.size(); i++) {
+    def split = splits[i]
+
+    testGroups["split${i}"] = {
+      node {
+        checkout scm
+
+        sh 'mvn clean -B -V -U -e'
```

Comment:
Thanks for detecting that!  I didn't understand why the "mvn" tool was not being auto-installed on the machines that were running tests.  I think your comment may explain that observed behavior, since the function was not called to compute the mvn tool on that node.